File: SpringBoot-Simple-JPA/src/main/java/com/example/springbootsimplejpa/service/OrderService.java
package com.example.springbootsimplejpa.service;

import com.example.springbootsimplejpa.domain.OrderModel;
import com.example.springbootsimplejpa.repository.OrderRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class OrderService {
    @Autowired
    private OrderRepository orderRepository;

    public OrderModel saveOrder(OrderModel orderDto) {
        return orderRepository.save(orderDto);
    }

    public List<OrderModel> saveOrderList(List<OrderModel> orderDtoList) {
        return orderRepository.saveAll(orderDtoList);
    }

    public List<OrderModel> getAllOrder() {
        return orderRepository.findAll();
    }

    public Optional<OrderModel> getOrderById(int id) {
        return orderRepository.findById(String.valueOf(id));
    }

    public Optional<OrderModel> getOrderByCode(String orderCode) {
        return Optional.ofNullable(orderRepository.findByOrderCode(orderCode));
    }
}

File: SpringKotlin-Simple-Example/src/main/kotlin/com/example/springkotlinsimpleexample/service/OrderService.kt
package com.example.springkotlinsimpleexample.service

import com.example.springkotlinsimpleexample.domain.dto.CreateOrderModelDTO
import com.example.springkotlinsimpleexample.domain.dto.ReadOrderModelDTO
import com.example.springkotlinsimpleexample.repository.OrderRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import javax.transaction.Transactional

@Component
class OrderService {

    @Autowired
    lateinit var orderRepository: OrderRepository

    fun getAllOrders(): List<ReadOrderModelDTO> {
        val orders = orderRepository.findAll()
        return orders.map { it.getReadOrderDTO() }
    }

    fun getOrderByOrderUserName(orderUserName: String): List<ReadOrderModelDTO> {
        val order = orderRepository.findAllByOrderUserName(orderUserName)
        return order.map { it.getReadOrderDTO() }
    }

    @Transactional
    fun createOrder(order: CreateOrderModelDTO): CreateOrderModelDTO {

        val requestOrderCode: String = order.orderCode
        val oldOrderCode: String = orderRepository.findByOrderCode(requestOrderCode)?.orderCode ?: ""

        orderRepository.findByOrderCode(requestOrderCode).let {
            if (it != null) {
                throw IllegalArgumentException("Order Code is already exist")
            }
        }

        return orderRepository.save(order.toEntity()).createOrderModelDTO()
    }
}
